pkg/asset/installconfig/azure/validation.go

```diff
@@ -87,25 +93,46 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	// Default requirements need to be sufficient to support Control Plane instances.
-	defaultInstanceReq := controlPlaneReq
-
-	if ic.ControlPlane != nil && ic.ControlPlane.Platform.Azure != nil && ic.ControlPlane.Platform.Azure.InstanceType != "" {
-		// Default requirements can be relaxed when the controlPlane type is set explicitly.
-		defaultInstanceReq = computeReq
+	defaultDiskType := aztypes.DefaultDiskType
+	defaultInstanceType := ""
 
-		allErrs = append(allErrs, ValidateInstanceType(client, field.NewPath("controlPlane", "platform", "azure"), ic.Azure.Region, ic.ControlPlane.Platform.Azure.InstanceType, controlPlaneReq)...)
+	if ic.Platform.Azure.DefaultMachinePlatform != nil && ic.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskType != "" {
+		defaultDiskType = ic.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskType
 	}
-
 	if ic.Platform.Azure.DefaultMachinePlatform != nil && ic.Platform.Azure.DefaultMachinePlatform.InstanceType != "" {
-		allErrs = append(allErrs, ValidateInstanceType(client, field.NewPath("platform", "azure", "defaultMachinePlatform"), ic.Azure.Region, ic.Platform.Azure.DefaultMachinePlatform.InstanceType, defaultInstanceReq)...)
+		defaultInstanceType = ic.Platform.Azure.DefaultMachinePlatform.InstanceType
+	}
+
+	if ic.ControlPlane != nil && ic.ControlPlane.Platform.Azure != nil {
+		controlPlaneDiskType := defaultDiskType
+		controlPlaneInstanceType := defaultInstanceType
+
+		if ic.ControlPlane.Platform.Azure.OSDisk.DiskType != "" {
+			controlPlaneDiskType = ic.ControlPlane.Platform.Azure.OSDisk.DiskType
+		}
+		if ic.ControlPlane.Platform.Azure.InstanceType != "" {
+			controlPlaneInstanceType = ic.ControlPlane.Platform.Azure.InstanceType
+		} else if controlPlaneInstanceType == "" {
+			controlPlaneInstanceType = defaults.ControlPlaneInstanceType(ic.Azure.Region)
+		}
```

Comment:
Re-arrange this to make it more clear what the precedence order is. I recommend using a similar structure for the disk type, too.
```suggestion
		controlPlaneInstanceType := ic.ControlPlane.Platform.Azure.InstanceType
		if controlPlaneInstanceType == "" {
			controlPlaneInstanceType = defaultInstanceType
		}
		if controlPlaneInstanceType == "" {
			controlPlaneInstanceType = defaults.ControlPlaneInstanceType(ic.Azure.Region)
		}
```